### PR TITLE
Revert "push post-release changelog and go.mod updates straight to master"

### LINKED ISSUE #15554: Release workflow that pushes post-release changelog and go.mod updates straight to master fails
This change was introduced with https://github.com/pulumi/pulumi/pull/15515

The workflow failed for the v3.108.0 release: https://github.com/pulumi/pulumi/actions/runs/8106099187/job/22155509142#step:4:2355

```
You are not currently on a branch.
Please specify which branch you want to merge with.
See git-pull(1) for details.

    git pull <remote> <branch>
```

The problem is we're not currently on a branch. The `actions/checkout@v3` checked us out at the `v3.108.0` tag.

However, even if we fix this up, I think we're still going to have a problem when we try to push straight to `master`.

I ran into the following errors when I was trying to push a commit straight to `master` to unblock the release:

Screenshot 2024-02-29 at 9 50 21 PM

I had temporarily added myself here:

Screenshot 2024-02-29 at 10 44 02 PM

However, it was still erroring that changes must be made through the merge queue and required status check "ci-ok" is expected.

I had to temporarily uncheck this to be able to push straight to `master`:

Screenshot 2024-02-29 at 10 44 49 PM

---

Until we have a solution, I'm thinking we may want to revert https://github.com/pulumi/pulumi/pull/15515 and also remove the code that tags `pkg/v${PULUMI_VERSION}`, and just do the `pkg/<version>` tag manually as part of the release process. And then come up with a different approach or automation to automatically create the tag once the `go .mod updates` commit lands in master. There may be a way we can automate adding the tag after the change lands in master, perhaps if the PR has a certain label or is named a certain way? But will we be able to push a new tag even with the "Do not allow bypassing the above settings" checkmark is checked?

### PR DESCRIPTION
**Pull request by justinvp**
This PR reverts the change from https://github.com/pulumi/pulumi/pull/15515 because it's currently broken, and even if we fix the problem where it is broken, I'm not sure we can make it work without unchecking the "Do not allow bypassing the above settings" in `master`'s branch protection rules, which I think we do want to keep checked (some more details in #15554).

An additional commit also temporarily disables tagging `pkg/<version>` in the workflow. Instead, that will have to be done manually as part of the release process when the change lands in `master`. As a fast follow, I'd like to see if there's a way we can automate adding the tag after the change lands in master, perhaps if the PR has a certain label or is named a certain way?

I mainly want to do this because it looks like we may have a Node.js SDK regression and will need to do a quick v3.108.1 release, and I want to make sure we can get that out without a broken workflow and needing to go through a bunch of manual temporary workarounds to unblock that release.

@tgummerer, what do you think?

Fixes #15554

(If we merge this, we should re-open https://github.com/pulumi/pulumi/issues/15458 to track fixing this)

### CODE REVIEW

**Comment by justinvp** [comment]
Test failures due to https://github.com/pulumi/pulumi/issues/15556

**Review by tgummerer — APPROVED** [review]
Yes, let's revert so we can get the next release out and push the tag manually (it will have to be done after the PR created here is merged).  I'll look into doing the tagging post merge separately, but I don't want to hold up the release for that, so this looks like the best option for now.

**Comment by julienp** [comment]
Rebased and forced pushed to pick up the test skips from https://github.com/pulumi/pulumi/pull/15544

**Pull request merged**
